ddtrace/appsec/_asm_context.py

```diff
@@ -0,0 +1,59 @@
+import contextlib
+from typing import Any
+
+from ddtrace.vendor import contextvars
+
+
+# FIXME: remove these and use the new context API once implemented and allowing
+# contexts without spans
+
+_DD_EARLY_IP_CONTEXTVAR = contextvars.ContextVar("datadog_early_ip_contextvar", default=None)
+_DD_EARLY_HEADERS_CONTEXTVAR = contextvars.ContextVar("datadog_early_headers_contextvar", default=None)
+_DD_EARLY_HEADERS_CASE_SENSITIVE_CONTEXTVAR = contextvars.ContextVar(
+    "datadog_early_headers_casesensitive_contextvar", default=False
+)
+
+
+def reset():
+    _DD_EARLY_IP_CONTEXTVAR.set(None)
+    _DD_EARLY_HEADERS_CONTEXTVAR.set(None)
+    _DD_EARLY_HEADERS_CASE_SENSITIVE_CONTEXTVAR.set(False)
+
+
+def set_ip(ip):  # type: (str) -> None
+    _DD_EARLY_IP_CONTEXTVAR.set(ip)
+
+
+def get_ip():  # type: () -> str
+    return _DD_EARLY_IP_CONTEXTVAR.get()
+
+
+def set_headers(headers):  # type: (Any) -> None
```

Comment:
I used `Any` because different frameworks use different objects for headers and it's really opaque to these functions since they just store or retrieve them without changing them or iterating or reading the internal data so it also doesn't make sense to force them into a dict or Mapping at this stage like the `trace_utils` do later (since they need to read and iterate). I'll add a comment to explain this.